src/ripple/app/misc/impl/Manifest.cpp

```diff
@@ -421,11 +433,12 @@ ManifestCache::applyManifest(Manifest m)
             return ManifestDisposition::badMasterKey;
         }
 
+        assert(m.signingKey || m.revoked());
```

Comment:
We're in a function that returns the disposition of the manifest.  I think that a better solution than a `assert` would be to actually perform the check and return the appropriate response.  Perhaps something like this inside of the immediately following condition...
```
        if (!revoked)
        {
            if (!m.signingKey)
            {
                JLOG(j_.warn()) << to_string(m)
                                << ": is not revoked but has no signing key";
                return ManifestDisposition::invalid;
            }

            // Sanity check: the ephemeral key of this manifest should not be
            ...
```